### PR TITLE
Fixes for converting an ERC-20 token to an NFT

### PR DESCRIPTION
**Pull request by bergeron**
## Explanation

Fixes 2 issues when converting an ERC-20 token to an NFT:
- The contract address was not pre-populated in the modal
- The NFT was not removed from the tokens tab

This may have been a regression.  The NFT import modal already looked for a contract address + token removal indicator in `window.history`, but callers [1](https://github.com/MetaMask/metamask-extension/pull/19806/files#diff-2296a08979d8dc2053c60cc8e7c883d2054fc02592cf1e790f274c16e7cdd29fL42) and [2](https://github.com/MetaMask/metamask-extension/pull/19806/files#diff-4d87baa5c1671beca2baa2d8a456d56a581908e26d8acddc3a3fb9a4b413f1b5L318) stopped setting the `history` state.  Instead of `history`, I am now using redux state to pass data to the modal.  But I'm open to other suggestions for passing data to modals.

### Before

During import:

https://github.com/MetaMask/metamask-extension/assets/3500406/89a93706-c80f-4e03-9b95-cb1745f1610d

Already imported (+ NFT remains on tokens tab):

https://github.com/MetaMask/metamask-extension/assets/3500406/7af96407-cc5d-44a1-b3cb-a7ba33daf263

### After

During import:

https://github.com/MetaMask/metamask-extension/assets/3500406/6ece1d55-0ee6-4789-9ecd-bb320036a493

Already imported (+ NFT removed from tokens tab):

https://github.com/MetaMask/metamask-extension/assets/3500406/e07c56f5-9abc-4707-b60e-6e8a5fb21ca6

## Manual Testing Steps

Try to import an NFT contract address as a token.  When prompted to convert to an NFT, the address should be pre-populated.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

### CODE REVIEW

File: ui/ducks/app/app.ts [hunk 1]
```diff
@@ -26,7 +26,12 @@ interface AppState {
     values?: { address?: string | null };
   } | null;
   networkDropdownOpen: boolean;
-  importNftsModalOpen: boolean;
+  importNftsModal: {
+    open: boolean;
+    tokenAddress?: string;
+    tokenId?: string;
+    ignoreErc20Token?: boolean;
+  };
```
Comment (bergeron) [inline ui/ducks/app/app.ts:34]: any backwards compatibility concerns with changing the state schema?

**Comment by bergeron** [comment]
Investigating test failure

```xml
<testcase
  classname="ERC721 NFTs testdapp interaction"
  name="should prompt users to add their NFTs to their wallet (one by one)"
  time="37.8">
  <failure>
    Waiting for element to be located By(xpath, //*[contains(text(), 'TestDappNFTs (6)')])
    Wait timed out after 10015ms
    TimeoutError: Waiting for element to be located By(xpath, //*[contains(text(), 'TestDappNFTs (6)')])
    Wait timed out after 10015ms
  </failure>
</testcase>
```

**Review by pedronfigueiredo — APPROVED** [review]

**Review by sahar-fehri — APPROVED** [review]

File: ui/components/multichain/import-nfts-modal/import-nfts-modal.js [hunk 2]
```diff
@@ -63,17 +63,14 @@ export const ImportNftsModal = ({ onClose }) => {
...
   const chainId = useSelector(getCurrentChainId);
-  const addressEnteredOnImportTokensPage =
-    history?.location?.state?.addressEnteredOnImportTokensPage;
-  const contractAddressToConvertFromTokenToNft =
-    history?.location?.state?.tokenAddress;
+  const {
+    tokenAddress: initialTokenAddress,
+    tokenId: initialTokenId,
+    ignoreErc20Token,
+  } = useSelector((state) => state.appState.importNftsModal);
```
Comment (salimtb) [inline ui/components/multichain/import-nfts-modal/import-nfts-modal.js:70]: the function `(state) => state.appState.importNftsModal)` can be extracted to the file selectors.js and be imported , like what we're doing for the line above

**Review by darkwing — APPROVED** [review]

File: ui/components/multichain/import-nfts-modal/import-nfts-modal.js (hunk 2 above)
Comment (bergeron) [reply ui/components/multichain/import-nfts-modal/import-nfts-modal.js:70]: My take is that a function isn't needed for one line that's called in 1 place.  I think it adds more code to maintain without benefit.  I *would* do a function if there was logic to encapsulate, or it was reused in several places.

**Pull request merged**